File: src/main/java/com/svantle/score/commands/HealCommand.java
package com.svantle.score.commands;

import com.svantle.score.utils.MessagesHandler;

import org.bukkit.Bukkit;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

public class HealCommand implements CommandExecutor {

    private final MessagesHandler messagesHandler;

    public HealCommand(MessagesHandler messagesHandler) {
        this.messagesHandler = messagesHandler;
    }

    @Override
    public boolean onCommand(CommandSender sender, Command cmd, String label, String[] args) {
        if (args.length == 0) {
            if (!(sender instanceof Player player)) {
                sender.sendMessage(messagesHandler.prefixed(messagesHandler.get("only_players")));
                return true;
            }

            if (!player.hasPermission("score.heal")) {
                player.sendMessage(messagesHandler.prefixed(messagesHandler.get("no_permission")));
                return true;
            }

            heal(player);
            player.sendMessage(messagesHandler.prefixed(messagesHandler.get("heal_success")));
            return true;
        }

        else if (args.length == 1) {
            if (!sender.hasPermission("score.heal.others")) {
                sender.sendMessage(messagesHandler.prefixed(messagesHandler.get("no_permission")));
                return true;
            }

            Player target = Bukkit.getPlayer(args[0]);
            if (target == null) {
                sender.sendMessage(messagesHandler.prefixed(messagesHandler.get("player_not_found")));
                return true;
            }

            heal(target);
            target.sendMessage(messagesHandler.prefixed(messagesHandler.get("healed")));

            sender.sendMessage(messagesHandler.prefixed(messagesHandler.get("healed_others", "player", target.getName())));
            return true;
        }

        else {
            sender.sendMessage(messagesHandler.prefixed(messagesHandler.get("heal_usage")));
            return true;
        }
    }

    private void heal(Player player) {
        player.setHealth(player.getAttribute(org.bukkit.attribute.Attribute.GENERIC_MAX_HEALTH).getValue());
    }
}

File: src/main/kotlin/com/svantle/score/commands/HealCommand.kt
package com.svantle.score.commands

import com.svantle.score.SCore
import com.svantle.score.constants.*
import com.svantle.score.utils.MessageHelper
import com.svantle.score.utils.sendTo
import org.bukkit.Bukkit
import org.bukkit.attribute.Attribute.GENERIC_MAX_HEALTH
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class HealCommand(core: SCore) : CommandExecutor {

	init {
		core.getCommand("heal")!!.setExecutor(this)
	}

	override fun onCommand(sender: CommandSender, cmd: Command, label: String, args: Array<String>?): Boolean {
		if(args.isNullOrEmpty()) {
			handleSelfHeal(sender)
		} else {
			handleHealOthers(sender, args[0])
		}

		return true
	}

	private fun handleSelfHeal(sender: CommandSender) {
		if(sender !is Player) {
			MessageHelper.phrase(MSG_ONLY_PLAYERS).sendTo(sender)
			return
		}

		if(sender.hasPermission(HEAL_COMMAND)) {
			heal(sender)
		} else {
			MessageHelper.phrase(MSG_NO_PERMISSION).sendTo(sender)
		}
	}

	private fun handleHealOthers(sender: CommandSender, targetName: String) {
		if (sender.hasPermission(HEAL_COMMAND_OTHERS) || sender !is Player) {
			val target = Bukkit.getPlayer(targetName)

			if(target == null) {
				MessageHelper.phrase(MSG_PLAYER_NOT_FOUND, targetName).sendTo(sender)
				return
			}

			heal(target, sender)
		} else {
			MessageHelper.phrase(MSG_NO_PERMISSION).sendTo(sender)
		}
	}

	private fun heal(player: Player, executor: CommandSender? = null) {
		player.health = player.getAttribute(GENERIC_MAX_HEALTH)!!.value

		if(executor != null) {
			MessageHelper.phrase(MSG_HEAL_SUCCESS_OTHERS, player.name).sendTo(executor)
		}

		MessageHelper.phrase(MSG_HEAL_SUCCESS).sendTo(player)
	}
}
